libsplinter/src/circuit/template/yaml_parser/v0_1.rs

```diff
@@ -0,0 +1,80 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[derive(Deserialize, Debug, Clone)]
+pub struct YamlCircuitCreateTemplate {
```

Comment:
Does the name of this struct need to be prefixed with `Yaml`? I think this is implied by the module it's in (`yaml_parser`). Same with the other structs here.

Reply:
I don't think its necessarily bad to have the prefix given the potential overlap with the non-parsing structs of similar name. (I'm not advocating strongly for keeping it, but we probably don't want multiple 'CircuitTemplate' structs.

Reply:
Addressed

Reply:
I changed it to CircuitCreateTemplate, in this case all the structs will have to be referred by their version too, `v1::CircuitCreateTemplate`, so it should be easy enough to differentiate